code/modules/antagonists/vampire/vampire_powers/umbrae_powers.dm

```diff
@@ -106,6 +106,89 @@
 	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/effect/proc_holder/spell/vampire/soul_anchor
+	name = "Soul Anchor (30)"
+	desc = "You summon a dimenional anchor after a delay, recasting will teleport you back to the anchor. you are forced back after 2 minutes."
+	gain_desc = "You have gained the ability to save a point in space and teleport back to it at will, do beware after saving you are forced back to that point after 2 minutes."
+	required_blood = 30
+	centcom_cancast = FALSE
+	base_cooldown = 3 MINUTES
+	action_icon_state = "shadow_anchor"
+	should_recharge_after_cast = FALSE
+	deduct_blood_on_cast = FALSE
+	var/obj/structure/shadow_anchor/anchor
+	var/timer
+
+/obj/effect/proc_holder/spell/vampire/soul_anchor/create_new_targeting()
+	return new /datum/spell_targeting/self
+
+/obj/effect/proc_holder/spell/vampire/soul_anchor/cast(list/targets, mob/user)
+	if(timer && !anchor) // second cast, but we are impatient
+		to_chat(user, "<span class='notice'>Your anchor isn't ready yet!</span>")
+		return
+
+	if(!timer && !anchor) // first cast, setup the anchor
+		var/turf/anchor_turf = get_turf(user)
+		timer = addtimer(CALLBACK(src, .proc/make_anchor, user, anchor_turf), 10 SECONDS, TIMER_STOPPABLE)
```

Comment:
This should probably be a `do_after` and have a message so the user has feedback that something is happening.